oauth.go

```diff
@@ -0,0 +1,82 @@
+package twilio
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/golang-jwt/jwt"
+	"github.com/twilio/twilio-go/client"
+	iam "github.com/twilio/twilio-go/rest/iam/v1"
+)
+
+type TokenAuth struct {
+	token string
+	OAuth client.OAuth
+}
+
+func (t *TokenAuth) NewTokenAuth(token string, o client.OAuth) *TokenAuth {
+	return &TokenAuth{token: token, OAuth: o}
+}
+
+func (t *TokenAuth) FetchToken() (string, error) {
+	if t.token != "" && !t.TokenExpired() {
+		return t.token, nil
+	}
+
+	token, err := t.OAuth.GetAccessToken(context.Background())
+	if err != nil {
+		return "", err
+	}
+
+	t.token = token
+	return t.token, nil
+}
+
+func (t *TokenAuth) TokenExpired() bool {
+	token, _, err := new(jwt.Parser).ParseUnverified(t.token, jwt.MapClaims{})
+	if err != nil {
+		return true
+	}
+
+	if claims, ok := token.Claims.(jwt.MapClaims); ok {
+		if exp, ok := claims["exp"].(float64); ok {
+			expirationTime := time.Unix(int64(exp), 0)
```

Comment:
I expect that Twilio will give back a timestamp in UTC, but `time.Unix` will use whatever the local time zone is, which could be something different.

Can you add a test for this behavior where the local time differs from UTC, both forwards and backwards, and verify that the expiration still behaves as expected in each case?